src/ast.c

```diff
@@ -92,3 +92,34 @@ static void ast_print_rec(const struct ast *ast, const int id, const int level)
 void ast_print(const struct ast *ast) {
 	ast_print_rec(ast, 0, 0);
 }
+
+static void ast_dot_rec(const struct ast *ast, FILE *stream, const size_t id) {
+	const struct ast_node *node = &ast->nodes[id];
+
+	fprintf(stream, "\t%zu [label = \"", id);
+
+	if (node->type < 256) {
+		fprintf(stream, "\"%c\"", (char)node->type);
+	} else {
+		fprintf(stream, "%s", token_to_str(node->type));
+	}
```

Comment:
I wonder if it's worth to create a `node_to_str(enum yytokentype token)` function to encapsulate this logic.

Maybe even just modify the already existing `token_to_str(enum yytokentype token)` to deal with literal tokens. 

But I'm not sure if we are ever needing to print nodes again, so maybe it's just unnecessary to create the new function. Thoughts?

Reply:
This would need to allocate memory, or add a LUT.
Maybe if we start to have this logic in more places, we could do that on the future.
Also, maybe in the future we could stop using the bison tokens on the AST and don't have the chars as a token option.